src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/TaskExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace System.Threading.Tasks
+{
+    internal static class TaskExtensions
+    {
+        public static async Task OrTimeout(this Task task, TimeSpan timeout)
+        {
+            var completed = await Task.WhenAny(task, Task.Delay(timeout));
+            if(completed == task)
+            {
+                // Manifest any exception
+                task.GetAwaiter().GetResult();
```

Comment:
Yeah, it just seemed unnecessary to use the sugar again. I can switch if you'd think it'd be clearer